kiali/kiali
# Use Istiod Registry in Kiali validations

Fixes https://github.com/kiali/kiali/issues/4027

This PR fetches the ServiceRegistry from the IstioD component to be used as an additional source in the Kiali validations.

The goal is not to change how validations are processed but enrich the cases where the registry can complement the information needed to validate the hosts.

When @xeviknal is back we can discuss more changes on this area but for this PR this PR will fix the "false errors" in validations due the remove services (exported).

How to test:
- First, check that the Maistra federation scenarios are covered https://github.com/maistra/istio/tree/maistra-2.1/pkg/servicemesh/federation/example/config-poc (configurations with no "false validation errors").
- Second, check that similar scenarios defined in multi-cluster upstream are covered https://istio.io/latest/docs/setup/install/multicluster/ (configurations with no "false validations errors").
- Third, a regression test in validation features.
- Forth, there is an additional fetch to the IstioD registry that is cached, so it shouldn't introduce any perf penalty, but worth to test if there is any change in that aspect.
